provider/maas/maas2instance.go

```diff
@@ -54,10 +55,25 @@ func (mi *maas2Instance) Id() instance.Id {
 }
 
 func (mi *maas2Instance) Addresses() ([]network.Address, error) {
-	machineAddresses := mi.machine.IPAddresses()
-	addresses := make([]network.Address, len(machineAddresses))
-	for i, address := range machineAddresses {
-		addresses[i] = network.NewAddress(address)
+	subnetsMap, err := mi.environ.subnetToSpaceIds()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	// Get all the interface details and extract the addresses.
+	interfaces, err := maas2NetworkInterfaces(mi, subnetsMap)
+
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	var addresses []network.Address
+	for _, iface := range interfaces {
+		if iface.Address.Value != "" {
+			logger.Debugf("found address %q on interface %q", iface.Address, iface.InterfaceName)
```

Comment:
This should probably be Trace since it would be a log line for every address on every machine whenever we check it.